harness/determined/cli/trial.py

```diff
@@ -121,6 +127,81 @@ def trial_logs(args: Namespace) -> None:
     )
 
 
+@authentication.required
+def generate_support_bundle(args: Namespace) -> None:
+    try:
+        output_dir = args.output_dir
+        if output_dir is None:
+            output_dir = os.getcwd()
+
+        dt = datetime.now().strftime("%Y%m%dT%H%M%S")
+        tar_filename = f"det-bundle-trial-{args.trial_id}-{dt}.tar.gz"
+        fullpath = os.path.join(output_dir, tar_filename)
+
+        with tempfile.TemporaryDirectory() as temp_dir, tarfile.open(fullpath, "w:gz") as bundle:
+            trial_logs_filepath = write_trial_logs(args, temp_dir)
+            master_logs_filepath = write_master_logs(args, temp_dir)
+            api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)
+
+            bundle.add(
+                trial_logs_filepath,
+                arcname=os.path.join(tar_filename, os.path.basename(trial_logs_filepath)),
+            )
+            bundle.add(
+                master_logs_filepath,
+                arcname=os.path.join(tar_filename, os.path.basename(master_logs_filepath)),
+            )
+            bundle.add(
+                api_trail_filepath,
+                arcname=os.path.join(tar_filename, os.path.basename(api_trail_filepath)),
+            )
+            bundle.add(
+                api_experiment_filepath,
+                arcname=os.path.join(tar_filename, os.path.basename(api_experiment_filepath)),
+            )
+
+            print(f"bundle path: {fullpath}")
```

Comment:
Sorry if I was confusing in my last comment.

Putting the files inside of a "directory" whose name is also the name of the `tar.gz` file it is, causes a problem when extracting. Because it tries to write the files into a path as if the `tar.gz` file was a directory. See:

```
(dev-3.7.11) [ttucker@gir tmp]$ tar zxvf det-bundle-trial-11-20220420T085359.tar.gz
det-bundle-trial-11-20220420T085359.tar.gz/trial_logs.txt
tar: det-bundle-trial-11-20220420T085359.tar.gz/trial_logs.txt: Cannot open: Not a directory
det-bundle-trial-11-20220420T085359.tar.gz/master_logs.txt
tar: det-bundle-trial-11-20220420T085359.tar.gz/master_logs.txt: Cannot open: Not a directory
det-bundle-trial-11-20220420T085359.tar.gz/api_trial_call.json
tar: det-bundle-trial-11-20220420T085359.tar.gz/api_trial_call.json: Cannot open: Not a directory
det-bundle-trial-11-20220420T085359.tar.gz/api_experiment_call.json
tar: det-bundle-trial-11-20220420T085359.tar.gz/api_experiment_call.json: Cannot open: Not a directory
tar: Exiting with failure status due to previous errors
```


This is what I put in my last review:

```
    dt = datetime.now().strftime("%Y%m%dT%H%M%S")
    bundle_name = f"det-bundle-trial-{args.trial_id}-{dt}"
    fullpath = os.path.join(output_dir, f"{bundle_name}.tar.gz")

    with tempfile.TemporaryDirectory() as temp_dir, tarfile.open(fullpath, "w:gz") as bundle:
        trial_logs_filepath = write_trial_logs(args, temp_dir)
        master_logs_filepath = write_master_logs(args, temp_dir)
        api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)

        bundle.add(trial_logs_filepath,
                   arcname=os.path.join(bundle_name, os.path.basename(trial_logs_filepath)))
        bundle.add(master_logs_filepath,
                   arcname=os.path.join(bundle_name, os.path.basename(master_logs_filepath)))
        bundle.add(api_trail_filepath,
                   arcname=os.path.join(bundle_name, os.path.basename(api_trail_filepath)))
        bundle.add(api_experiment_filepath,
                   arcname=os.path.join(bundle_name, os.path.basename(api_experiment_filepath)))

        print(f"bundle path: {fullpath}")
```

We need to make a `bundle_name` which will become the directory that the files get extracted into. The name of the `tar.gz` file should be the `bundle_name` with `tar.gz` at the end.

I usually find it helpful to run the these features from the point of view of different personas, Sales Engineers, Support, Customers, etc. and in various scenarios, single trial, HP search, huge logs, etc. and it helps me to spot these kinds of issues / improvements.

Reply:
done!